apis/ec2/manualv1alpha1/instances_types.go

```diff
@@ -0,0 +1,437 @@
+
+	// The metadata options for the instance. For more information, see Instance
+	// Metadata and User Data (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html).
+	// +optional
+	MetadataOptions *InstanceMetadataOptionsRequest `json:"metadataOptions"`
+
+	// Specifies whether detailed monitoring is enabled for the instance.
+	// +optional
+	Monitoring *RunInstancesMonitoringEnabled `json:"monitoring,omitempty"`
+
+	// The network interfaces to associate with the instance. If you specify a network
+	// interface, you must specify any security groups and subnets as part of the
+	// network interface.
+	// +optional
+	NetworkInterfaces []InstanceNetworkInterfaceSpecification `json:"networkInterfaces,omitempty"`
+
+	// The placement for the instance.
+	// +optional
+	Placement *Placement `json:"placement,omitempty"`
+
+	// [EC2-VPC] The primary IPv4 address. You must specify a value from the IPv4
+	// address range of the subnet.
+	//
+	// Only one private IP address can be designated as primary. You can't specify
+	// this option if you've specified the option to designate a private IP address
+	// as the primary IP address in a network interface specification. You cannot
+	// specify this option if you're launching more than one instance in the request.
+	//
+	// You cannot specify this option and the network interfaces option in the same
+	// request.
+	// +optional
+	PrivateIPAddress *string `json:"privateIpAddress,omitempty"`
+
+	// The ID of the RAM disk to select. Some kernels require additional drivers
+	// at launch. Check the kernel requirements for information about whether you
+	// need to specify a RAM disk. To find kernel requirements, go to the AWS Resource
+	// Center and search for the kernel ID.
+	//
+	// AWS recommends that you use PV-GRUB instead of kernels and RAM disks. For more
+	// information, see PV-GRUB (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/UserProvidedkernels.html)
+	// in the Amazon Elastic Compute Cloud User Guide.
+	// +optional
+	RAMDiskID *string `json:"ramDiskId,omitempty"`
+
+	// Region is the region you'd like your Instance to be created in.
+	Region *string `json:"region"`
+
+	// The IDs of the security groups. You can create a security group using CreateSecurityGroup
+	// (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateSecurityGroup.html).
+	//
+	// If you specify a network interface, you must specify any security groups
+	// as part of the network interface.
+	// +optional
+	SecurityGroupIDs []string `json:"securityGroupIds,omitempty"`
+
+	// SecurityGroupIDRefs is a list of references to SecurityGroups used to set
+	// the SecurityGroupIDs.
+	// +optional
+	SecurityGroupIDRefs []xpv1.Reference `json:"securityGroupIdRefs,omitempty"`
+
+	// SecurityGroupIDsSelector selects references to SecurityGroupID used
+	// to set the SecurityGroupIDs.
+	// +optional
+	SecurityGroupIDSelector *xpv1.Selector `json:"securityGroupIdSelector,omitempty"`
```

Comment:
Do we need both these and the SecurityGroupRefs and SecurityGroupSelectors below?

Reply:
I don't believe so. I this was just me following a pre-existing pattern and applying it in too many places. Is there a preference for which one stays and which one goes?

Reply:
I've removed the ones highlighted above. Let me know @AaronME if you'd rather those stay and the others go.